### ISSUE TITLE
Configurable Starters

### ISSUE THREAD

**Issue by dwieeb** [ionic-team/ionic-cli#3499]
Many framework CLIs have a single starter with feature flags to enable/disable/switch certain features and preferences such as TypeScript, testing frameworks, theme, etc. We've already seen requests for this, such as https://github.com/ionic-team/ionic-cli/issues/3490. 

The starters in place for Ionic 4 beta are the typical Ionic-style starters, but can we make one starter where the contents and options are configurable? Can `ionic start` prompt for all these options?

Examples:

```
ionic start --tabs
ionic start --no-karma
ionic start --theme=dark
ionic start --no-lazy-loading
```

**Comment by rdlabo** [comment]
I think this is possible.. For examle, this. https://github.com/ionic-team/ionic-cli/pull/3477 
After git clone, CLI change templates is not difficult.

**Comment by dwieeb** [comment]
I'm not certain how scalable that is. If the starter's file structure changes even slightly, it could result in runtime errors and will be difficult to maintain. 

Your PR for the `-s` flag could easily be a part of this new type of starter, however:

```
ionic start --no-karma --feature-structure=type
```

^ This would start a new app without karma and the old way of structuring features (pages, components, etc).

**Comment by rdlabo** [comment]
I'm glad for v3's folder structure can also be used in v4!
May I try to challenge implementation?

**Comment by dwieeb** [comment]
A note: the important conceptual change here is that the logic of how to structure a starter is shifted onto the starter template itself, and the Ionic CLI becomes just a way to ask for inputs and write files. This makes `ionic start` more bullet-proof and future-proof, on top of unlocking a bunch of features. 

cc @johneast

**Comment by Sampath-Lokuge** [comment]
Hi, @dwieeb Will we have this feature soon on the Ionic 5 app? i.e. create CSS apps. Because of @adamdbradley  instruct us to use CSS only apps. It is here: https://twitter.com/adamdbradley/status/1250790908833955840

**Comment by dwieeb** [comment]
Probably not soon, no. I wouldn't make decisions for your organization based on opinions on Twitter. Use what works best for you and your team. It isn't terribly difficult to change an Ionic Angular app to use CSS if that's what you want.

**Comment by Sampath-Lokuge** [comment]
@dwieeb `it isn't terribly difficult to change an Ionic Angular app to use CSS if that's what you want.` Yes, what is the best way to do that? Are you telling me to change each and every SCSS file manually after it was generated by the CLI or else?

**Comment by dwieeb** [comment]
> Are you telling me to change each and every SCSS file manually after it was generated by the CLI

Yes. You could also start with an Angular app (with the css option) and run the `ng add @ionic/angular`, but I haven't tried that myself.

**Comment by Sampath-Lokuge** [comment]
@dwieeb 
> You could also start with an Angular app (with the css option) and run the ng add @ionic/angular

I don't think this is a good option since we cannot create Side menu, Tabs and Capacitor apps by default here. It will be a big loss for us. 

The manual change also an annoying thing since we need to change the file extension and its usage file and etc. 

So our only hope here is you'll give this soon for us!

**Comment by dwieeb** [comment]
Yeah, there isn't an ideal workaround right now. Hopefully I can prioritize configurable starters soon.

**Comment by Sampath-Lokuge** [comment]
@dwieeb Thanks a lot. We'll highly appreciate it since now CSS is the default template for almost all the top frameworks. That is manually due to CSS supports variables and a lot more now and evolving it rapidly.

**Comment by eliashdezr** [comment]
Any progress on this?
